src/ui/wxWidgets/properties.cpp

```diff
@@ -422,10 +429,68 @@ wxIcon CProperties::GetIconResource( const wxString& WXUNUSED(name) )
  * wxEVT_COMMAND_BUTTON_CLICKED event handler for wxID_OK
  */
 
-void CProperties::OnOkClick( wxCommandEvent& /* evt */ )
+void CProperties::OnOkClick( wxCommandEvent& WXUNUSED(evt) )
 {
 ////@begin wxEVT_COMMAND_BUTTON_CLICKED event handler for wxID_OK in CProperties.
   // Before editing this code, remove the block markers.
   EndModal(wxID_OK);
 ////@end wxEVT_COMMAND_BUTTON_CLICKED event handler for wxID_OK in CProperties.
 }
+
+/*!
+ * wxEVT_COMMAND_BUTTON_CLICKED event handler for wxID_CHANGE_NAME
+ */
+void CProperties::OnEditName( wxCommandEvent& WXUNUSED(evt) )
+{
+  wxTextEntryDialog textInputDialog(
+    this, _("Name:"), _("Please enter the new database name"), m_NewDbName.c_str(), wxOK|wxCANCEL
+  );
+
+  textInputDialog.SetSize(550, -1);
+
+  if (textInputDialog.ShowModal() == wxID_OK) {
+    m_DbName = textInputDialog.GetValue();
+
+    if (m_DbName.IsEmpty()) { /* Show 'N/A' on the UI in case of an empty string, */
+      m_DbName = _("N/A");    /* but use the empty string as new DB name          */
+
+      if (Validate() && TransferDataToWindow()) {
+        m_NewDbName = std2stringx(_T(""));
+      }
+    }
+    else {
+      if (Validate() && TransferDataToWindow()) {
+        m_NewDbName = std2stringx(m_DbName.wc_str());
+      }
+    }
+  }
+}
+
+/*!
+ * wxEVT_COMMAND_BUTTON_CLICKED event handler for wxID_CHANGE_DESCRIPTION
+ */
+void CProperties::OnEditDescription( wxCommandEvent& WXUNUSED(evt) )
```

Comment:
The implementation follows the Windows version, the reference implementation, where for each editable property a single button exists at the outmost right side of the dialog.

![image](https://user-images.githubusercontent.com/4042917/45614015-5f1e8180-ba68-11e8-9086-fffb8d4760a5.png)

In my opinion this results in a user expectation that a separat dialog, specific to the property, gets shown. If there should be one dialog for all editiable properties then there should be only one "Edit" button located somewhere else at this dialog, but this would provide a deviation from the reference implementation. The current solution makes use of some simple standard dialogs which the wx framework provides. Your proposal would make it necessary to create a custom dialog, for currently just two editable properties. But I agree, the more editiable properties exists, the more such a custom dialog makes sense.

Reply:
Makes sense.